source/parser/intermediate-representator.cpp

```diff
@@ -24,6 +24,21 @@
 #include "parser/memory-allocator.hpp"
 #include "parser/symbol-table.hpp"
 
+void IntermediateRepresentator::generateMacroInformation(
+    FinalRepresentation& representation) {
+  for (const auto& i : _commandList) {
+    if (dynamic_cast<IntermediateMacroInstruction*>(i.get()) == nullptr)
```

Comment:
`dynamic_cast`s are very expensive in c++, because the runtime must traverse the entire inheritance tree to make sure these classes are related. Could we do this differently (maybe, in the future, use tags in the classes, i.e. enum members)